pkg/operator/sync_openshiftcontrollermanager_v311_00.go

```diff
@@ -303,41 +323,47 @@ func manageRouteControllerManagerClientCA_v311_00_to_latest(client coreclientv1.
 	return resourceapply.SyncConfigMap(context.Background(), client, recorder, util.KubeAPIServerNamespace, apiserverClientCA, util.RouteControllerTargetNamespace, apiserverClientCA, []metav1.OwnerReference{})
 }
 
-// similar logic for route-controller-manager in manageRouteControllerManagerConfigMap_v311_00_to_latest
-func manageOpenShiftControllerManagerConfigMap_v311_00_to_latest(clusterVersionInformer configinformersconfigv1.ClusterVersionInformer, kubeClient kubernetes.Interface, client coreclientv1.ConfigMapsGetter, recorder events.Recorder, operatorConfig *operatorapiv1.OpenShiftControllerManager) (*corev1.ConfigMap, bool, error) {
-	clusterVersion, err := clusterVersionInformer.Lister().Get("version")
-	if err != nil {
-		return nil, false, err
-	}
+// groupDisabledControllers gets caps from configv1.ClusterVersion and populates
+// groupDisabled with all controllers that are known in the ClusterVersion and
+// also disabled.
+func groupDisabledControllers(clusterVersion *configv1.ClusterVersion) map[controlplanev1.OpenShiftControllerName]bool {
 	knownCaps := sets.New[configv1.ClusterVersionCapability](clusterVersion.Status.Capabilities.KnownCapabilities...)
 	capsEnabled := sets.New[configv1.ClusterVersionCapability](clusterVersion.Status.Capabilities.EnabledCapabilities...)
 
-	groupDisabled := make(map[string]bool)
-	if knownCaps.Has(configv1.ClusterVersionCapabilityDeploymentConfig) && !capsEnabled.Has(configv1.ClusterVersionCapabilityDeploymentConfig) {
-		groupDisabled["openshift.io/deploymentconfig"] = true
+	groupDisabled := make(map[controlplanev1.OpenShiftControllerName]bool)
+	for _, cont := range knownControllers {
+		if knownCaps.Has(ControllerCapabilities[cont]) && !capsEnabled.Has(ControllerCapabilities[cont]) {
```

Comment:
`if ControllerCapabilities[cont] != "" && knownCaps.Has(ControllerCapabilities[cont]) && !capsEnabled.Has(ControllerCapabilities[cont]) {` will safe the sets from checking for an empty string.